### PR TITLE
Added utils dir

### PR DESCRIPTION
**Pull request by VytalyGorbatov**
Added utils directory with 3-rd party JSON library and template for access synchronization.

### CODE REVIEW

File: utils/include/synchronized.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#ifndef SYNCHRONIZED_H_INCLUDED
+#define SYNCHRONIZED_H_INCLUDED
+
+#pragma once
+
+#include <mutex>
+using namespace std;
```
Comment (AntonVoytov) [inline utils/include/synchronized.h:7]: Do not pollute the namespace.

File: utils/include/synchronized.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#ifndef SYNCHRONIZED_H_INCLUDED
```
Comment (AntonVoytov) [inline utils/include/synchronized.h:1]: Prefer use pragma once only

File: utils/include/synchronized.h [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#define SYNCHRONIZED_H_INCLUDED
+
+#pragma once
+
+#include <mutex>
+using namespace std;
+
+template <typename T>
+class Synchronized {
```
Comment (AntonVoytov) [inline utils/include/synchronized.h:10]: Please rename class

File: utils/include/synchronized.h [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    T& ref_to_value;
+    lock_guard<mutex> guard;
+  };
+
+  Access GetAccess() {
+    return {value, lock_guard(m)};
+  }
+
+private:
+  T value;
```
Comment (AntonVoytov) [inline utils/include/synchronized.h:27]: Maybe if the field was a reference type would that be a better option? Also, the constructor should be changed

**Review by AntonVoytov — CHANGES_REQUESTED** [review]
Please refactor this file

**Comment by VytalyGorbatov** [comment]
@AntonVoytov  , please, rereview this request with the changes made.

**Pull request closed without merging**